app/models/manageiq/providers/network_manager.rb

```diff
@@ -54,9 +54,10 @@ class << model_name
                :class_name  => "ManageIQ::Providers::BaseManager",
                :autosave    => true
 
+    has_many :availability_zones, :primary_key => :parent_ems_id, :foreign_key => :ems_id
```

Comment:
Nice idea! 👍 

There is one catch however.

What if `parent_ems_id=nil`, then we would get possibly all disconnected availability_zones.

Usually, NetworkManager must have parent_ems (my assumption reading the code). So, this ^^ should not happen. Just for sure, can we add validatation on parent_ems_id not to be nil?